src/main/java/mesosphere/marathon/client/MarathonClient.java

```diff
@@ -27,11 +33,33 @@ public Exception decode(String methodKey, Response response) {
 	}
 	
 	public static Marathon getInstance(String endpoint) {
-		GsonDecoder decoder = new GsonDecoder(ModelUtils.GSON);
-		GsonEncoder encoder = new GsonEncoder(ModelUtils.GSON);
-		return Feign.builder().encoder(encoder).decoder(decoder)
+		return getInstance(endpoint, null);
+	}
+
+	/**
+	 * The generalized version of the method that allows more in-depth customizations via
+	 * {@link RequestInterceptor}s.
+	 *
+	 * @param endpoint
+	 * 		URL of Marathon
+	 */
+	public static Marathon getInstance(String endpoint, RequestInterceptor... interceptors) {
+		Builder b = Feign.builder()
+				.encoder(new GsonEncoder(ModelUtils.GSON))
+				.decoder(new GsonDecoder(ModelUtils.GSON))
 				.errorDecoder(new MarathonErrorDecoder())
-				.requestInterceptor(new MarathonHeadersInterceptor())
-				.target(Marathon.class, endpoint);
+				.requestInterceptor(new MarathonHeadersInterceptor());
+
+		if (interceptors!=null)
+			b.requestInterceptors(asList(interceptors));
```

Comment:
requestInterceptors overwrite existing requestInterceptors, so the MarathonHeadersInterceptor gets removed in the process.